src/jit/compiler.hpp

```diff
@@ -1405,7 +1408,7 @@ inline
 void                GenTree::InitNodeSize(){}
 
 inline
-void                GenTree::SetOper(genTreeOps oper)
+void                GenTree::SetOper(genTreeOps oper, bool clearVN)
 {
     gtOper = oper;
```

Comment:
I don't think we have tried building the JIT without SMALL_TREE_NODES defined in quite sometime.
But you should go ahead and add the if (clearVM) logic here as well as it appears to just be missing.
The instance field gtVNPair is available with and without SMALL_TREE_NODES.